src/main/compatibility-check.ts

```diff
@@ -6,8 +7,14 @@ import rimraf from 'rimraf'
 import {app, dialog} from 'electron'
 import {DATADIR_VERSION, COMPATIBLE_VERSIONS} from '../shared/version'
 import {config} from '../config/main'
-import {mainLog} from './logger'
-import {TFunctionMain} from './i18n'
+import {createAndInitI18nForMain, createTFunctionMain} from './i18n'
+import {DEFAULT_LANGUAGE} from '../shared/i18n'
+
+export type DatadirChecked = 'DATADIR_CHECKED'
```

Comment:
This should be something more sophisticated (like branded type from `io-ts`) to simply make it impossible to create value with that type outside this module, or to make it troublesome enough so it can be immediately spotted on review